src/sidebars/favourites/favourites.js

```diff
@@ -2,117 +2,126 @@
-                }
+                    // Add new stocks from favourites
+                    self.favourites.map(function(favourite) {
+                        if (self.stocks.map(function(stock) { return stock.code; }).indexOf(favourite) === -1) {
+                            // This is a new stock
+                            quandlService.getData(favourite, function(stock) {
+                                var data = stock.data[0],
+                                    price,
+                                    delta,
+                                    percentage;
+
+                                if (data) {
+                                    price = data.close;
+                                    delta = data.close - data.open;
+                                    percentage = delta / data.open * 100;
+
+                                    self.stocks.push({
+                                        name: stock.name,
+                                        code: stock.code,
+                                        price: price,
+                                        delta: delta,
+                                        percentage: Math.abs(percentage),
+                                        notification: false,
+                                        favourite: true
+                                    });
+                                }
+                            });
+                        }
+                    })
+                };
 
-                // Add new stocks from favourites
-                self.favourites.map(function(favourite) {
-                    if (self.stocks.map(function(stock) { return stock.code; }).indexOf(favourite) === -1) {
-                        // This is a new stock
-                        quandlService.getData(favourite, function(stock) {
-                            var data = stock.data[0],
-                                price,
-                                delta,
-                                percentage;
-
-                            if (data) {
-                                price = data.close;
-                                delta = data.close - data.open;
-                                percentage = delta / data.open * 100;
-
-                                self.stocks.push({
-                                    name: stock.name,
-                                    code: stock.code,
-                                    price: price,
-                                    delta: delta,
-                                    percentage: Math.abs(percentage),
-                                    notification: false
-                                });
-                            }
-                        });
-                    }
-                })
-            };
+                self.update();
 
-            self.updateFavourites();
-        }])
+                $scope.$on('favouriteChanged', function(event, data) {
```

Comment:
When the icon is clicked we need to
a) remove the local storage entry
b) remove the favourite from the search bar
c) remove the favourite from the favourites

The only ways I can think of doing all of these are to
a) call several functions from the html ng-click on different (unrelated?) controllers
b) emit an event and catch it in the relevant places.

I opted for the event as I felt it was a valid use of it, and leads to tidier code -- I hope! And to do that we need to use `$scope`.

Can you think of another way to either use an event, or not use the event at all?